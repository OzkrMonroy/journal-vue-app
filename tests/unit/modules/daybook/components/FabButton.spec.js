import { shallowMount } from "@vue/test-utils";
import FabButton from "@/modules/daybook/components/FabButton";

describe("Tests for the FabButton", () => {
  const wrapper = shallowMount(FabButton);

  test("should render the component correctly", () => {
    expect(wrapper.html()).toMatchSnapshot();
  });

  test("should set the fa-plus value when the icon property is not sent", () => {
    const icon = wrapper.find("i").classes("fa-plus");

    expect(icon).toBeTruthy();
  });

  test("should set the value that is sent as a property", () => {
    const wrapper = shallowMount(FabButton, {
      props: {
        icon: "fa-delete",
      },
    });
    const icon = wrapper.find("i").classes("fa-delete");
    const defaultIcon = wrapper.find("i").classes("fa-plus");

    expect(icon).toBeTruthy();
    expect(defaultIcon).toBeFalsy();
  });

  test("should emit the on:click event when the button is clicked", () => {
    wrapper.find("button").trigger("click");
    expect(wrapper.emitted("on:click")).toHaveLength(1);
  });
});
