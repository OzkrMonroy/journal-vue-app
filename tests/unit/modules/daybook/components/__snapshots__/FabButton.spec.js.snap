// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tests for the FabButton should render the component correctly 1`] = `<button class="btn btn-primary"><i class="fa fa-2x fa-plus"></i></button>`;
